postupgrade/utils.js

```diff
@@ -5,15 +5,77 @@ const simpleGit = require('simple-git/promise')();
 
 /**
  * Merges two comment-json files, with the original having higher priority.
- * @param {string} updatedCommentJson 
- * @param {string} originalCommentJson 
+ * Prunes away duplicated commented out props.
+ *
+ * @param {string} newJson 
+ * @param {string} originalJson 
  * @returns {string}
  */
-exports.mergeJson = function(updatedCommentJson, originalCommentJson) {
-  const merged = assign(parse(updatedCommentJson), parse(originalCommentJson));
+exports.mergeGlobalConfigs = function (newJson, originalJson) {
+  const originalParsed = parse(originalJson);
+  const commentsFromOriginal = parseCommentedOutProps(originalParsed);
+  const newPruned = pruneDuplicatedComments(parse(newJson), commentsFromOriginal);
+  const merged = assign(newPruned, originalParsed);
   return stringify(merged, null, 2);
 }
 
+/**
+ * Parses "commented out" config values.
+ *
+ * @param {import('comment-json').CommentJSONValue} jsonWithComments
+ * @returns {import('comment-json').CommentToken[]}
+ */
+function parseCommentedOutProps(commentJsonValue) {
+  return getPropCommentSymbols(commentJsonValue).flatMap(symbol => {
+    const commentArr = commentJsonValue[symbol] || [];
+    // We only care about non-inline LineComments that "look like" a config option
+    return commentArr.filter(c => {
+      return !c.inline && c.type === 'LineComment' && c.value.match(/^\s?"\w+":/)
+    })
+  });
+}
+exports.parseCommentedOutProps = parseCommentedOutProps;
+
+/**
+ * Removes all comments in CommentJSONValue that have the same value and type as the given
+ * commentsToDedupe.
+ *
+ * @param {import('comment-json').CommentJSONValue} commentJsonValue
+ * @param {import('comment-json').CommentToken[]} commentsToDedupe
+ * @returns {import('comment-json').CommentJSONValue} the updated value
+ */
+function pruneDuplicatedComments(commentJsonValue, commentsToDedupe) {
```

Comment:
Small nit: Should we just call this `pruneComments(parsedJson, comments)`? That might be a bit simpler. If I understand correctly, the purpose of this method is to ensure the provided comments are removed from the `parsedJson`. This method isn't concerned with why those comments need to be removed (e.g. they are dupes), just that they are removed.

Reply:
yup! I'll remove all instances of words like dedupe from this method